Issue/2040 configure merge policy (#2045)
quickwit-oss/quickwit: docs/configuration/index-config.md

| ------------- | ------------- | ------------- |
| `timestamp_field`      | Timestamp field used for sharding documents in splits (1).   | None |
| `commit_timeout_secs`      | Maximum number of seconds before committing a split since its creation.   | 60 |

> | `split_num_docs_target` | Target of number of docs per split.   | 10_000_000 |

Reviewer: ```suggestion
| `split_num_docs_target` | Target number of docs per split.   | 10_000_000 |
```

> | `merge_policy` | Describes the strategy used to trigger split merge operations (see [Merge policies] section below). |

Reviewer: ```suggestion
| `merge_policy` | Describes the strategy used to trigger split merge operations (see [Merge policies](#merge-policies) section below). |
```

| `resources.heap_size`      | Indexer heap size per source per index.   | 2_000_000_000 |

(1) Both `datetime` and `i64` can be referenced. `i64` fields are interpreted as Unix timestamp (seconds). You can learn more about time sharding [here](./../concepts/architecture.md).

### Merge policies

Quickwit makes it possible to define the strategy used to decide which splits should be merged together and when.

Quickwit offers three different merge policies, each with their
own set of parameters.

#### "Stable log" merge policy

The stable log merge policy attempts to minimize write amplification AND keep time-pruning power as high as possible, by merging splits with a similar size, and with a close time span.

> Quickwit's default merge policy is a stable log merge policy
> with the following parameter:

Reviewer 2: ```suggestion
Quickwit's default merge policy is the `stable_log` merge policy
with the following parameters:
```

```yaml
version: 0
index_id: "hdfs"
# ...
indexing_settings:
  merge_policy:
    type: "stable_log"
    min_level_num_docs: 100_000
    merge_factor: 10
    max_merge_factor: 12
```

| Variable      | Description   | Default value |
| ------------- | ------------- | ------------- |
| `merge_factor`      | *(advanced)* Number of splits to merge together in a single merge operation.   | 10 |
| `max_merge_factor` | *(advanced)* Maximum number of splits that can be merged together in a single merge operation.  | 12 |
| `min_level_num_docs` |  *(advanced)* Number of docs below which all splits are considered as belonging to the same level.   | 100_000 |

#### "Limit Merge" merge policy

*The limit merge policy is considered advanced*.

The limit merge policy simply limits write amplification by setting an upperbound
of the number of merge operation a split should undergo.

```yaml
version: 0
index_id: "hdfs"
# ...
indexing_settings:
  merge_policy:
    type: "limit_merge"

>     min_level_num_docs: 100_000

Reviewer: Did you mean?
```suggestion
    max_merge_ops: 5
```

Author: yes thank you

    merge_factor: 10
    max_merge_factor: 12

```

| Variable      | Description   | Default value |
| ------------- | ------------- | ------------- |
| `max_merge_ops`   |  Maximum number of merges that a given split should undergo. | 4 |
| `merge_factor`      | *(advanced)* Number of splits to merge together in a single merge operation.   | 10 |
| `max_merge_factor` | *(advanced)* Maximum number of splits that can be merged together in a single merge operation.  | 12 |

#### No merge

The `no_merge` merge policy entirely disables merging.

> This setting is not recommended.
> Merges are necessary to reduce the number of splits,
> and hence improve search performances.

Reviewer: ```suggestion
:::caution
This setting is not recommended. Merges are necessary to reduce the number of splits,
and hence improve search performances.
:::
```
